### PR TITLE
cleanup sql functions

### PR DESCRIPTION
**Pull request by nguyenvukhang**
and add second-class support for PostgreSQL

### CODE REVIEW

**Comment by nguyenvukhang** [comment]
this also fixes issue #35

File: packages/database/src/api/sql/reset/module.ts [hunk 1]
```diff
@@ -1,9 +1,12 @@
-import { remove } from '../../../sql/remove'
 import { config } from '../../../config'
+import { sql } from '../../../sql'
 
-remove.tables(config.database, [
-  'user_modules_done_module',
+sql.dropTables(config.database, [
+  'dag_modules_placed_module',
+  'dag_modules_hidden_module',
+  'dag_modules_hidden_module',
```
Comment (weiseng18) [inline packages/database/src/api/sql/reset/module.ts:7]: duplicate table

File: packages/database/tests/environment.ts [hunk 2]
```diff
@@ -1,22 +1,23 @@
...
+  await sql.restoreFromFile(database, config.restoreSource)
 }
 
 /**
  * post-test teardown
  * @param {string} database
  */
 export async function teardown(database: string) {
-  await remove.database(database)
+  // await remove.database(database)
```
Comment (weiseng18) [inline packages/database/tests/environment.ts:19]: remove?

File: packages/database/tests/module-pull.test.ts [hunk 3]
```diff
@@ -17,10 +17,13 @@ afterAll(() => teardown(dbName))
 
 jest.setTimeout(60000)
 test('pull all modules from NUSMods', async () => {
-  await remove.tables(dbName, [
-    'degree_modules_module',
-    'user_modules_done_module',
+  await sql.dropTables(dbName, [
+    'dag_modules_placed_module',
+    'dag_modules_hidden_module',
+    'dag_modules_hidden_module',
```
Comment (weiseng18) [inline packages/database/tests/module-pull.test.ts:23]: duplicate table

**Pull request closed without merging**